### DISCUSSION TITLE
Strange remote: warning re libjfxwebkit.dylib when pushing to GitHub

### DISCUSSION THREAD

**Discussion by abulka** [wiverson/maven-jpackage-template#48] [Q&A] [answered]
When pushing a copy of this repository to a new repository for the first time, I'm getting a strange

    remote: warning: File macos-javafx/javafx-sdk-15.0.1/lib/libjfxwebkit.dylib is 80.17 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB

even though there is no such file in the repository or working directory?

Repro:
- clone this repository
- git remote rename origin original-wiverson
- git remote add origin git@github.com:YOUR_USER_NAME/maven-jpackage-template.git
- git branch -M main
- git push -u origin main

```
Enumerating objects: 798, done.
...
remote: Resolving deltas: 100% (403/403), done.
remote: warning: See http://git.io/iEPt8g for more information.
remote: warning: File macos-javafx/javafx-sdk-15.0.1/lib/libjfxwebkit.dylib is 80.17 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: warning: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To github.com:YOUR_USER_NAME/maven-jpackage-template.git
 * [new branch]      main -> main
Branch 'main' set up to track remote branch 'main' from 'origin'.
```

The git push seems to succeed but in the terminal on my Mac, `find . -name libjfxwebkit.dylib` finds nothing, and searching for `libjfxwebkit` in the source finds nothing?

**Comment by wiverson** [comment, marked as answer]
So, when I originally set up the repository as a regular repository, I included the webkit libraries. Then, when I flipped it to a template, GitHub complained that files greater than 50mb are too big for templates. Because of how Git works, you're getting a copy of that old library in the history even though it's no longer used.

The error message you are seeing is basically boils down to GitHub complaining about a large file in the history and recommending you install Git LFS. If you use something like GitHub Desktop it can set up Git LFS for you on your repository with a click, or you can look up how to do it at the command line.

**Reply by abulka** [reply]
Ah, I see - thanks for the explanation!

Presumably I could delete the git history `rm -f .git` and initialise my copy of the repo again with `git init` to clear out those old files, before pushing to my own repository. There's probably a smarter way using git but I think that would do the trick.

**Reply by wiverson** [reply]
Totally. AFAIK there is also a way to purge old stuff from a repo but TBH it looks like more work and also hacking on a repo like that seems like a great way to get to a potentially corrupted state. YMMV.
